src/readers/include/idata_provider.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "../../common/include/shapes.h"
+#include "../../core/src/databases.h"
+class IDataProvider {
```

Comment:
Please explain in details why do we need IDataProvider and IReader at the same time? 
As I understand we need it because we have several types of schemes (cores) and we whant separate reader for each data type. 
If it is true, than 
1. We need to place IdataProvider inside core and move IReader to reader module.
2. IdataProvider should have function which takes file name and enum(should be placed in common) with scheme(core) type and will return the needed impl of itself.